tools/ManifestSignerClient/Readme.md

```diff
@@ -0,0 +1,44 @@
+2. Next choice is the different algorithmic parameters. For ECDsa, there are three EC parameters to choose from secp521r1 , secp384r1, prime256v1. Edit the parameter `ROOT_KEY_ALGO` and `SIGNER_KEY_ALGO`. Default value for root CA is `scep521r1` and signer cert is `prime256v1`
+3. Next choice is the different SHA algorithm. For RSA and ECDsa, there are three choices of SHA algorithm to choose from  SHA256, SHA384, SHA512. Recommendation is to use larger keys for root CA and smaller keys for signer certs. Edit the parameter `ROOT_SHA_ALGORITHM` and `SIGNER_SHA_ALGORITHM`. Default value is `SHA256`
+3. The file names of the root CA, root private key, signer cert, signer private key are also editable in the script. 
+
+Once the variables are set in the script. Run the script in a `Powershell` and the following files will be generated as a result.
+
+1. `root_ca_private_<algo>_key.pem` - private root key
+2. `root_ca_public_<algo>_cert.pem` - public root key / root CA
+3. `root_ca_public_<algo>_key.srl` - root .srl file
+4. `signer_private_<algo>_key.pem` - private signer key
+5. `signer_public_<algo>_key.pem` - prublic signer key
+5. `signer_public_<algo>_cert.pem` - public signer cert
+6. `signer_<algo>.csr` - signer .csr file
+
+The files that are needed to launch and sign deployment manifest JSON is as follows. 
+1. `root_ca_public_<algo>_cert.pem` - public root key / root CA
+2. `signer_private_<algo>_key.pem` - private signer key
+3. `signer_public_<algo>_cert.pem` - public signer cert
+
+### Step 2: Launch Settings
+Manifest Signer client has  `launchSettings.json` under `Properties` folder and the following values have to be set to sign the deployment manifest file. 
+1. `MANIFEST_VERSION` is the version of the deployment manifest JSON. Each time a new deployment file is generated, the manifest version has to be incremented. To begin, give `1` as the value. 
+2. `DSA_ALGORITHM` is the DSA algorithm scheme. The values supported for ECDsa are `ES256`, `ES384` and `ES512`. For RSA, it is `RS256`, `RS384` and `RS512`
+3. `DEPLOYMENT_MANIFEST_FILE_PATH` is the absolute path of the deployment manifest to be signed including the file name. 
+4. `SIGNED_DEPLOYMENT_MANIFEST_FILE_PATH` is the absolute path of the signed deployment manifest including the file name of your choice. 
+5. `MANIFEST_TRUST_DEVICE_ROOT_CA_PATH` is the absolute path of the file `root_ca_public_<algo>_cert.pem`
+6. `MANIFEST_TRUST_SIGNER_PRIVATE_KEY_PATH` is the absolute path of the file  `signer_private_<algo>_key.pem`
+7. `MANIFEST_TRUST_SIGNER_CERT_PATH` is the absolute path of the file `signer_public_<algo>_cert.pem` 
+
+### Step 3: Build and Run Manifest Signer Client
+Once the `launchSettings.json` file is configured, the solution can be built and run using `dotnet build` and `dotnet run`. If all the inputs are configured properly, then signed deployment JSON will be generated. 
```

Comment:
Are we planning to release the tool or customers will need to clone the code and build themselves?

Reply:
Right now setting it to build themselves. I need to understand the release plan for the binaries.